File: Armstrong Number/armstrongNumber__specialLogic.java
// Using Special Logic

import java.util.Scanner;
class ArmstrongNumber {
    public static void main(String[] args) {
        int x, q, r, sum=0, dividend;
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a Number : ");
        x = sc.nextInt();
        String y = String.valueOf(x);
        for(char i : y.toCharArray()){
            int j = Character.getNumericValue(i);
            sum = sum + j*j*j;
        }
        if(x==sum){
            System.out.print(x+" is an Armstrong Number");
        }else{
            System.out.print(x+" is Not an Armstrong Number");
        }
    }
}

File: Armstrong Number/armstrongNumber__specialLogic.py
# Using Special Logic

x=int(input("Enter a Number : "))
sum=0

for i in str(x):
    sum = sum + int(i)**3
    
if x==sum:
    print(f"{x} is an Armstrong Number")
else:
    print(f"{x} is Not an Armstrong Number")
